File: kafkaStreamsModelServerInMemoryStore/src/main/scala/com/lightbend/scala/naive/modelserver/PrintProcessor.scala
package com.lightbend.scala.naive.modelserver

import com.lightbend.scala.modelServer.model.ServingResult
import com.lightbend.scala.naive.modelserver.store.StoreState
import org.apache.kafka.streams.processor.AbstractProcessor


class PrintProcessor extends AbstractProcessor[Array[Byte], ServingResult]{

  private var modelStore: StoreState = null

  override def process (key: Array[Byte], value: ServingResult ): Unit = {
    value.processed match {
      case true => println(s"Calculated quality - ${value.result} calculated in ${value.duration} ms")
      case _ => println("No model available - skipping")
    }
  }
}

File: kafkaStreamsModelServerInMemoryStore/src/main/java/com/lightbend/java/naive/modelserver/PrintProcessor.java
package com.lightbend.java.naive.modelserver;

import com.lightbend.java.model.ServingResult;
import com.lightbend.java.naive.modelserver.store.StoreState;
import org.apache.kafka.streams.processor.AbstractProcessor;

/**
 * Created by boris on 7/12/17.
 */

public class PrintProcessor extends AbstractProcessor<byte[], ServingResult> {

    private StoreState modelStore;

    @Override
    public void process(byte[] key, ServingResult value) {
        if(value.isProcessed())
            System.out.println("Calculated quality - " + value.getResult() + " in " + value.getDuration() + "ms");
        else
            System.out.println("No model available - skipping");
    }
}
